starter-code/src/clue.js

```diff
@@ -1,94 +1,119 @@
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = {name: 'rope', weight: 10};
+var knife = {name: 'knife', weight: 8};
+var candlestick = {name: 'candlestick', weight: 2};
+var dumbbell = {name: 'dumbbell', weight: 30};
+var poison = {name: 'poison', weight: 2};
+var axe = {name: 'axe', weight: 15};
+var bat = {name: 'bat', weight: 13};
+var trophy = {name: 'trophy', weight: 25};
+var pistol = {name: 'pistol', weight: 20};
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var diningRoom = {name: 'Dining Room'};
+var conservatory = {name: 'Conservatory'};
+var kitchen = {name: 'Kitchen'};
+var study = {name: 'Study'};
+var library = {name: 'Library'};
+var billiardRoom = {name: 'Billiard Room'};
+var lounge = {name: 'Lounge'};
+var ballroom = {name: 'Ballroom'};
+var hall = {name: 'Hall'};
+var spa = {name: 'Spa'};
+var livingRoom = {name: 'Living Room'};
+var observatory = {name: 'Observatory'};
+var theater = {name: 'Theater'};
+var guestHouse = {name: 'Guest House'};
+var patio = {name: 'Patio'};
+
 
 // Characters Collection
 var charactersArray = [];
 
+charactersArray.push(mrGreen);
+charactersArray.push(drOrchid);
+charactersArray.push(profPlum);
+charactersArray.push(missScarlet);
+charactersArray.push(mrsPeacock);
+charactersArray.push(mrMustard);
+
 // Rooms' Collection
 var roomsArray = [];
 
+roomsArray.push(rope);
+roomsArray.push(knife);
+roomsArray.push(candlestick);
+roomsArray.push(dumbbell);
+roomsArray.push(poison);
+roomsArray.push(axe);
+roomsArray.push(trophy);
+roomsArray.push(pistol);
+
+
 // Weapons Collection
 var weaponsArray = [];
+
+weaponsArray.push(diningRoom);
+weaponsArray.push(conservatory);
+weaponsArray.push(kitchen);
+weaponsArray.push(study);
+weaponsArray.push(library);
+weaponsArray.push(lounge);
+weaponsArray.push(ballroom);
+weaponsArray.push(hall);
+weaponsArray.push(spa);
+weaponsArray.push(livingRoom);
+weaponsArray.push(observatory);
+weaponsArray.push(theater);
+weaponsArray.push(guestHouse);
+weaponsArray.push(patio);
+
+//Function randomSelector
+
+function randomSelector(array){
+    if (array.length === 0) {
+      return undefined;
+    } let rand = array[Math.floor(Math.random()*array.length)];
+      return rand;
+}
+
+//Function pickMistery
+
+function pickMistery(){
+    
+    var misteryEnvelope = [];
+
+    let character = randomSelector(charactersArray);
+    let weapon = randomSelector(weaponsArray);
+    let room = randomSelector(roomsArray);
+
+    misteryEnvelope.push(character);
+    misteryEnvelope.push(weapon);
+    misteryEnvelope.push(room);
+
+    return misteryEnvelope;
+
+}
+
+//Function revealMistery
+
+function revealMistery(misteryEnvelope) {
```

Comment:
Every variable takes part of memory so try not to use them when not necessary. You can:
```
return `${misteryEnvelope[0].first_name} ${misteryEnvelope[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].roomNames}!!!!`
```